Save ender chest when world is saved

Kamesuta opened the pull request:
## Description
- Modified the code to save Ender Chests when saving the world.
- By default, Ender Chests are only saved when a player logs out (unless the option to save when closing the chest is enabled).
- Since Ender Chests are not saved when the world is saved, their contents can be rolled back to the state of the last logged out player in the event of a server crash.
- This makes it possible to cause item loss by crashing the server with items in the chest, or item duplication by logging out with items in the chest.

## Changes
- Modified the code to save Ender Chests when saving the world.

## Related Issues
Resolves #130 

## Checklist
<!-- After posting your issue, please check the boxes below if they apply -->
- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.

utarwyn reviewed (commented):
Thank you for your contribution! The idea to save chests during the world save is very interesting, I totally approve this :+1:

But I have one question: do you think that deleting the context of players in the world during an automatic save can cause problems? (if they are interacting with their enderchest for example)

Kamesuta commented:
> Thank you for your contribution! The idea to save chests during the world save is very interesting, I totally approve this 👍

Thank you for your review and for approving the idea!


> But I have one question: do you think that deleting the context of players in the world during an automatic save can cause problems? (if they are interacting with their enderchest for example)

I have made some modifications to address this concern. Specifically:

- I removed the saving of context during times other than world save.
This is because saving context at other times can cause inconsistencies when items are added or removed from the ender chest, which can result in items being duplicated or lost.
The best solution is to only save context during world save.

- I modified the code to ensure that the container being used is dataized in the main thread to avoid inconsistencies.
This addresses the concern you raised about potential issues arising when players interact with their ender chest while dataization is performed asynchronously.
To do this, I first retrieve the container that the player has opened on the main thread, and then retrieve and serialize the other containers in a separate thread.

utarwyn commented:
Replaced by #261 as I can't push to your branch
